### ISSUE TITLE
Test has passed however Reason with an empty message is printed in the xml

### ISSUE THREAD

**Issue by psandhu79** [nunit/nunit#1896]
```
       <test-case id="0-2262" name="TestPickedPoint" fullname="A.B.C.TestPickedPoint" methodname="TestPickedPoint" classname="A.B.C" runstate="Runnable" seed="978314103" result="Passed" start-time="2016-11-08 15:28:16Z" end-time="2016-11-08 15:28:16Z" duration="0.010144" asserts="1">
                    <properties>
                      <property name="Category" value="SmokeTest" />
                    </properties>
                    <reason>
                      <message><![CDATA[]]></message>
                    </reason>
                  </test-case>
```

This when imported into TeamCity reports a failure

**Comment by rprouse** [comment]
The fact that TeamCity reports a passed test as a failure is a problem that should be fixed there. @NikolayPianikov any comments on that? What about `Assert.Pass("Message")`?

@psandhu79 can you give us a code example of a test that displays an empty message?

**Comment by psandhu79** [comment]
@rprouse Thank you looks like a bug in TeamCity found this 
https://youtrack.jetbrains.com/issue/TW-15717

However the tests pass in Nunit 2 for teamcity and fail in Nunit 3 i am guessing its a regression in the Nunit importer for the xml

**Comment by CharliePoole** [comment]
Even if it is not the cause of the TC errror, NUnit should not be creating an empty reason element under any circumstance. We need to confirm whether this happens, and when, and fix it.

Based on the TC issue, it appears to happen when AssertPass is used. We should check out Assert.Pass both with and without a message.

**Comment by psandhu79** [comment]
TC reports a failed test either with Assert.Pass() or Asset.Pass("No exception should occur until this point with current setup")

With a message the reason element contains the message

**Comment by CharliePoole** [comment]
There should only be a reason element if there is a reason. Of course, TC should deal with it anyway, but we should fix our output as well.

**Comment by NikolayPianikov** [comment]
@psandhu79 it looks like a bug in the TeamCity xml parser for NUnit. I will check it

**Comment by psandhu79** [comment]
@NikolayPianikov could you please update this bug 
https://youtrack.jetbrains.com/issue/TW-15717

If possible attached a new xml plugin and i can test

**Comment by NikolayPianikov** [comment]
@psandhu79 I can't reproduce this issue. Could you attach nunit xml report file?

**Comment by NikolayPianikov** [comment]
@psandhu79 Looks like this issue was fixed in TeamCity 10.0.3
So try upgrading to this version

**Comment by cristobalito** [comment]
I'll pick this one up - hopefully have a fix in the next few days.

**Comment by CharliePoole** [comment]
Great! Looking forward to seeing it.

**Comment by rprouse** [comment]
Fixed by #1955
